pkg/inventory/inventory-client.go

```diff
@@ -513,3 +541,46 @@ func (cic *ClusterInventoryClient) toInfo(obj *unstructured.Unstructured) (*reso
 func (cic *ClusterInventoryClient) helperFromInfo(info *resource.Info) *resource.Helper {
 	return resource.NewHelper(info.Client, info.Mapping)
 }
+
+func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured, dryRun common.DryRunStrategy) error {
```

Comment:
This doesn't seem right to me. Similar to how we put the decision on how/where the inventory list is added to a specific resource type in an type-specific implementation of the `Inventory` interface, this functionality should also belong in the type-specific implementation.

Reply:
The `Inventory` interface handles where and how the resource list is stored, it doesn't handle how the inventory object is applied to the cluster.
The `InventoryClient` handles how it is applied to the cluster. `updateStatus` is part of applying the inventory object to the cluster.

Reply:
The problem this is solving is that updating status must be a separate API call because the status is a subresource.

Unfortunately, the current abstraction has the InventoryClient in cli-utils making the API calls, which means it's not possible for the Inventory or InventoryInfo implementation to decide whether one or two API calls is needed.

This is the kludge I was talking about. To work around it would require more significant changes to these interfaces, like pushing the `applyInventoryObj` impl out of the InventoryClient and into something that can be injected.

Reply:
As discussed offline. I add this method as the minimum change using the current interface. Later, we should improve the interface for the inventory object. Then applying an inventory object is handled as part of the inventory interface.